src/workerd/jsg/modules.capnp

```diff
@@ -31,4 +31,7 @@ enum ModuleType {
 
   internal @2;
   # Provided by runtime but can only imported by builtin modules.
+
+  internalWasm @3;
```

Comment:
I'm not sure this really belongs in this enum -- I think this enum is expressing something that is orthogonal to the type of code. That is, it makes sense for a Wasm module to be any of `bundle`, `builtin`, or `internal`.

I would argue that instead, the `src` member of `struct Module` should become a union, like:

```
struct Module {
  # Javascript module with its source code.
  name @0 :Text;
  union {
    src @1 :Data;  # JS / TS code
    wasm @4 :Data;  # Wasm module
  }
  tsDeclaration @3 :Text;
  type @2 :ModuleType;
}
```

With that said, due to the lack of comments here I'm not entirely sure what `tsDeclaration` is and I may be getting this wrong...

Reply:
This union is a much better place for it. Requires a bit of rearrangement in `modules.h` to accommodate.

The vocabulary here is confusing. It might be a bit clearer to call this field `Visibility` or something similar since the three options are user, builtin public, builtin private. What's particularly weird here is that I think the word `bundle` in this "visibility" enum means the code is user code, but there is also this `struct Bundle` which is used for `BuiltinBundle`s like `NODE_BUNDLE` and is completely unrelated.

Looks to me like `tsDeclaration` is only used for rtti. I think it doesn't matter for any internal module.

Reply:
Okay I rearranged it to work this way and renamed "bundle" to "user" since I think that makes more sense.